config/shared/errors/errors.go

```diff
@@ -85,6 +85,7 @@ var (
 	ErrInvalidProxy              = errors.New("proxies must be http(s)")
 	ErrInsecureProxy             = errors.New("insecure plaintext HTTP proxy specified for HTTPS resources")
 	ErrPathConflictsSystemd      = errors.New("path conflicts with systemd unit or dropin")
+	ErrPathConflictsParentDir    = errors.New("path conflicts with parent directory of another file, link, or directory")
```

Comment:
Hmm, I wonder if this error is appropriate. 


From my understanding we are trying to have a validate error that indicates the user has likely messed up their config by accidentally labeling a Directory as a file, or by forgetting a Directory entry.

What do you think about `ErrMisslabeledDir= errors.New("parent directory path matches configured file, check path, and insure parent directory is configured")`

Not super sold on my error, but it is closer to what I would want to see as an enduser. (if my understanding of the issue is sound)

Reply:
Yes, I think this makes the error message clearer and more explicit to the end user. 

I was considering having more than one error message, instead of creating one generic message, creating one  for directory conflicts, another for files, and another for links. Is that a good practice?